OOModel/src/declarations/Class.cpp

```diff
@@ -228,4 +239,22 @@ Class* Class::expressionToClass(Expression* expr)
 	return nullptr;
 }
 
+QSet<Class*> Class::directSubClasses()
+{
+	auto top = root();
+	//Find all the classes where this method is subclasses
+	QSet<Class*> result;
+	QList<Model::Node*> toCheck;
+	toCheck.append(top);
+	while (!toCheck.isEmpty())
```

Comment:
Not sure but I think a Visitor which only visits class nodes and its sub declarations should not need to traverse the whole tree but only up to class nodes?
https://github.com/dimitar-asenov/Envision/blob/development/ModelBase/src/visitor/Visitor.h